envoy/config/grpc_mux.h

```diff
@@ -100,7 +100,7 @@ class GrpcMux {
   virtual GrpcMuxWatchPtr addWatch(const std::string& type_url,
                                    const absl::flat_hash_set<std::string>& resources,
                                    SubscriptionCallbacks& callbacks,
-                                   OpaqueResourceDecoder& resource_decoder,
+                                   OpaqueResourceDecoderSharedPtr& resource_decoder,
```

Comment:
Why are you passing a reference to the pointer here - if we're taking an ownership reference, shouldn't it be the share pointer itself?

Reply:
Oops, you are correct, thanks!
Fixed.